docs/concepts.rst

```diff
@@ -162,6 +162,31 @@ Example DAG with decorated style
         html_content=email_info['body']
     )
 
+DAG decorator
+-------------
+
+.. versionadded:: 2.0.0
+
+In addition to creating DAGs using context managed, in Airflow 2.0 you can also create DAGs from a function.
+DAG decorator creates a DAG generator function. This function when called returns a DAG.
+
+DAG decorator also sets up the parameters you have in the function as DAG params. This allows you to parametrize
+your DAGs and set the parameters when triggering the DAG manually. See
+:ref:`Passing Parameters when triggering dags <dagrun:parameters>` to learn how to pass parameters when triggering DAGs.
+
+Example DAG with decorator:
+
+.. exampleinclude:: /../airflow/example_dags/example_dag_decorator.py
+    :language: python
+    :dedent: 4
+    :start-after: [START dag_decorator_usage]
+    :end-before: [END dag_decorator_usage]
+
+.. note:: Note that Airflow will only load DAGs that appear in``globals()``. Which means you need to make sure to have
+  a variable for your returned DAG in the module scope. Otherwise Airflow won't detect your decorated DAG. In addition,
+  you may want to make your dag variable named ``DAG`` such that Airflow doesn't skip this module when the
+  ``DAG_DISCOVERY_SAFE_MODE`` is activated.
```

Comment:
Thoughts? I'm not sure how to run the system dags to make sure it runs CC @turbaszek

Reply:
Apart from this warning it would be good to add something like:
```
This will work:
dag = yout_dag_function()
but this will not:
yout_dag_function()
```

The system DAG should be runnable by `pytest tests/core/test_example_dags_system.py -k example_dag_decorator`

Reply:
Right now it would have to be `DAG = your_dag_function` -- and you also need the string `airflow` in the dag files somewhere `return all(s in content for s in (b'DAG', b'airflow'))` from  https://github.com/apache/airflow/blob/5fc06a6ef9c136a4861bc2426aea768e9a18dcbb/airflow/utils/file.py#L192-L211

Reply:
@casassg Has updated that check: https://github.com/apache/airflow/pull/10587/files#diff-d089cab1f4d8262ade0db0e327e7d094d2e8ecd592db597571a5c0a4a3d35ef8R213-R214